src/app/components/fruit-table/fruit-table.component.ts

```diff
@@ -8,11 +8,106 @@ import {FruitTableViewModel} from './fruit-table-view-model';
   providers: [FruitTableViewModel]
 })
 export class FruitTableComponent implements OnInit {
-  columnsToDisplay = ['id', 'name', 'genus', 'calories'];
+  columnsToDisplay = ['id', 'name', 'genus', 'calories', 'carbohydrates', 'sugar'];
 
   constructor(public viewModel: FruitTableViewModel) {
 
   }
+  dataSource = this.viewModel.fruitData$
+  filteredDataSource = this.dataSource.value||this.dataSource
```

Comment:
I would recommend putting the filteredDataSource is the ViewModel, right next to the fruitData$ declaration. You could also make it a BehaviorSubject just like the fruitData$. This lets you read from it asynchronously, which is nice because the data comes back from the API asynchronously.